### PR TITLE
Adding Google Authenticator 

### PR DESCRIPTION
**Pull request by anvyst**
I've expanded a bit the default LoginTrait to support Google Authenticator mechanism, by using [TwoFactorAuth](https://github.com/RobThree/TwoFactorAuth) plugin (since version  1.5.2, which became PSR-4 compatible for CakePHP3.x).

In order to function properly, there was also added migration file, to expand `users` table with `secret, secret_verified` fields. One holds Base32 shared secret, the second one - makes sure QR-code is shown in `verify.ctp` until the first successful verification/login into the system.

After the user is authorised from `/login`, I'm creating temporary session, with holds all required data for me, removing Auth.Users session (to avoid access for restricted url's), and redirect to `/verify`, where verification process takes place. Once verification is successful, I'm restoring Auth.Users session to let AuthComponent magic to work.

### CODE REVIEW

File: src/Controller/Traits/LoginTrait.php [hunk 1]
```diff
@@ -178,6 +180,84 @@ public function login()
...
+                        ->where(['id' => $user['id']])
+                        ->execute();
+                }
+
+                $this->request->session()->delete('temporarySession');
+                $this->request->session()->write('Auth.User', $user);
+
+                $url = $this->Auth->redirectUrl();
+
+                $this->redirect($url);
```
Comment (steinkel) [inline src/Controller/Traits/LoginTrait.php]: use return

File: src/Controller/Traits/LoginTrait.php [hunk 2]
```diff
@@ -178,6 +180,84 @@ public function login()
...
+            } else {
+                $message = __d('CakeDC/Users', 'Verification code is invalid. Try again');
+                $this->Flash->error($message, 'default', [], 'auth');
+
+                $this->request->session()->destroy();
+
+                $this->redirect(Configure::read('Auth.loginAction'));
```
Comment (steinkel) [inline src/Controller/Traits/LoginTrait.php]: use return

File: src/Template/Users/verify.ctp [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+            <div class="users form well well-lg">
+                <?= $this->Form->create() ?>
+
+                <?= $this->Flash->render('auth') ?>
+                <?= $this->Flash->render() ?>
+                <fieldset>
+                    <?php if (!empty($secretDataUri)):?>
+                        <p class='text-center'><img src="<?php echo $secretDataUri;?>"/></p>
+                    <?php endif;?>
+                    <?= $this->Form->input('code', ['required' => true,'label' => 'Verification Code']) ?>
```
Comment (steinkel) [inline src/Template/Users/verify.ctp]: use __d() for label

File: composer.json [hunk 4]
```diff
@@ -36,15 +37,17 @@
         "league/oauth2-instagram": "@stable",
         "league/oauth2-google": "@stable",
         "league/oauth2-linkedin": "@stable",
-        "google/recaptcha": "@stable"
+        "google/recaptcha": "@stable",
+        "robthree/twofactorauth": "^1.5.2"
```
Comment (anvyst) [inline composer.json]: @steinkel , I've removed it from `require` param, and left it only in `require-dev` just to make sure that unit tests will pass.

Comment (steinkel) [inline composer.json]: makes sense, thanks

**Pull request merged**